### ISSUE TITLE
Strings like 'l2' and 'l1' would be preferable as 'L2' and 'L1"

### ISSUE THREAD

**Issue by woodsp-ibm** [Qiskit/qiskit-machine-learning#82]
In skimming over the changes that black did I noticed the strings 'l1' and 'l2'. Normally for names lower case l is avoided as it can be mistaken for capital I and even for numeric 1 depending on font. I think it would be preferable to have these as uppercase L, since the docstring even talks about L2 loss where L is in capitals. This is in code like NetworkNeuralClassifier/Regressor among others. And where we test the value, if we choose to accept lower case for compatibility, then maybe do an upper() so it can be compared to 'L2' etc.

**Comment by adekusar-drl** [comment]
I agree in general, but we have other loss functions coded as lower case strings: `cross_entropy`, `cross_entropy_sigmoid`, so I think we need a consistent solution. I suggest replace `l1` with `absolute_error` and `l2` with `squared_error`.
@beichensinn what do you think?

**Comment by woodsp-ibm** [comment]
The strings across the API are case insensitive as it does a lower() on what is passed in to compare against constants in the code in lowercase. If upper() was done instead it would just mean the constants all became upper internally and maybe we'd show them that way in the docstring too. It would not change the API in this case; but changing to different strings means deprecating the l1 and l2 so they'd have to stay in the code for a while. Of course doing this upper() instead, and having it as 'L1' internally and in the docs, the API, would not prevent users from using the 'l1' etc in lowercase in their code and thus have the potential readability problem there still, that different strings would prevent.

**Comment by mattwright99** [comment]
@woodsp-ibm @adekusar-drl I am new to qiskit contributions but I am interested in picking up this issue. Would that be ok?

I think using `upper()` and comparing to `'L1'` internally is a good idea since it would make the code more readible and the docstrings more clear. I think it could also be beneficial to add support for `'absolute_error'` and `'squared_error'` in addition to `'L1'` and `'L2'` respectively.
I also saw a comment in #92 suggesting that we should extract the `if...else` block to a seperate class and I could see that being in the scope of this PR. Do you agree? If so, could you provide some more information on that (e.g. what class should I look at)?
Thanks!

**Comment by adekusar-drl** [comment]
@mattwright99 Surely, you can work on this issue!
I'd prefer to keep string constants in lower case just to keep them more readable in the documentation. I suggest to gracefully deprecate `l1` and `l2` in favor of 'absolute_error' and 'squared_error' cause this would solve the issue completely.
The `if` block you mentioned, is it the `if` block where we instantiate a loss object out of a string value? If so, then I happy to discuss your suggestion here, but please don't code it upfront.

**Comment by mattwright99** [comment]
@adekusar-drl Alright, i am excited to start contributing!
My plan at the moment is to change all the occurances of 'l1' and 'l2' to 'absolute_error' and 'squared_error' throughout the source code, the unit tests, and documentation. I will also continue to handle 'l1' and 'l2' as we do now but output a depreciation warning if they are provided. Should i use the `deprecate_arguments()` function? If so, could you explain how that works or point me to an example? Anything else I might be missing?

**Comment by manoelmarques** [comment]
@adekusar-drl I added all the possible ways you can use deprecations in the test_deprecation.py. However All methods/decorators expect you to deprecate a class, method, argument. property or function. Nothing for default values. The message being created automatically doesn't support it. Would it be possible for you to add a warnings.warn method in your PR for that situation and I will then create some method to support it and use it in place, either in the same PR or another ?

**Comment by adekusar-drl** [comment]
@manoelmarques Yes, we can generate manual warnings.
@mattwright99 Please add lines like these:
```python
warnings.warn(
    "The 'l2' value is deprecated as of 0.2.0 "
    "and will be removed no sooner than 3 months after the release. "
    "You should use the 'squared_error' value instead.",
    DeprecationWarning,
)
```
The version number in the text depends on the date when you'll be able to finish the PR. We can edit it later on.

**Comment by mattwright99** [comment]
@adekusar-drl sounds good, I am just about to publish a PR
